templates/prometheus.systemd.erb

```diff
@@ -7,7 +7,7 @@ After=basic.target network.target
 User=<%= scope.lookupvar('prometheus::server::user') %>
 Group=<%= scope.lookupvar('prometheus::server::group') %>
 ExecStart=<%= scope.lookupvar('prometheus::server::bin_dir') %>/prometheus \
-  <%= @daemon_flags.join(" \\\n  ") %> \
+  <%= @daemon_flags.join("\n  ") %>
   <%= scope.lookupvar('prometheus::server::extra_options') %>
```

Comment:
If `extra_options` is used (ie not an empty string), this will be broken?

Reply:
Yes. 

I'm using this fix:
```diff
- $daemon_flags = $daemon_flags_basic + "--web.external-url=${prometheus::server::external_url}"
+ $daemon_flags = "--web.external-url=${prometheus::server::external_url} \\" + $daemon_flags_basic
```

Reply:
Hi, yes, the fix proposal by @matejzero works

Reply:
Looking at it again, I think this still won't fix every usecase. Right not, it works if we are using `external_url`, but it will fail if we use `extra_options`, since there is no trailing `\` at the end of `daemon_flags`.

In the pre-PR code, it failed if `extra_options` were empty string, since there was a trailing `\` at the end.

I have a fix, should I open another PR or should I post the solution here?

Reply:
Can you propose another PR?

Reply:
Will do. I did some work yesterday, but it looks like it's gonna take some time to cover all init providers, not only systemd.

Reply:
How about merging `extra_options` and `daemon_flags` in the puppet code?